File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int m =0;
        int best = prices[0];
        for(int  i = 1; i<prices.size(); i++){
            if(prices[i]>best){
                m = max(m,prices[i]-best);
            }
            best = min(best,prices[i]);
        
        
        }
        return m;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] nums) {
        int n = nums.length;
        int best = nums[0];
        int mp = 0;
        for(int i = 1; i<n; i++){
            
            if(nums[i]>best){
                
                mp = Math.max(mp,(nums[i]-best));
            }
            best = Math.min(best,nums[i]);
            

        }
        return mp;

        
    }
}